project/src/main/webapp/closebrackets.js

```diff
@@ -0,0 +1,191 @@
+// CodeMirror, copyright (c) by Marijn Haverbeke and others
+// Distributed under an MIT license: https://codemirror.net/LICENSE
+
```

Comment:
what is this file for? Maybe add a comment at the top

Reply:
please add more documentation to this file (eg short descriptions on top of each function)

Reply:
Imported directly from codemirror to have brackets auto close